### PR TITLE
feat(set-net-error)

### PR DESCRIPTION
**Pull request by emberglaz**

### CODE REVIEW

File: app/components/AvatarList/AvatarList.tsx [hunk 1]
```diff
@@ -10,10 +12,15 @@ import { UserType } from '../../utils/types';
...
 import { AvatarListType, IAvatarListItem } from './AvatarList.data';
+import { showInternetErrorAlert } from '../../utils';
 
 export const AvatarList: React.FC<IAvatarListProps> = ({ data, parent }) => {
   const navigation = useNavigation();
   const dispatch = useAppDispatch();
+  const netInfo = useNetInfo();
+  const { t } = useTranslation();
+
+  const isConnected = useMemo(() => netInfo.isConnected, [netInfo.isConnected]);
```
Comment (Reykjavik151) [inline app/components/AvatarList/AvatarList.tsx]:
```suggestion
const { isConnected } = useNetInfo();

File: app/constants/error.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const NETWORK_ERROR_TITLE = 'No Internet connection';
+export const NETWORK_ERROR_DESCRIPTION =
+  'Please make sure you are connected to the internet and try again. ';
```
Comment (Reykjavik151) [inline app/constants/error.ts:3]: I18n?

File: app/hooks/quest.tsx [hunk 3]
```diff
@@ -28,6 +36,17 @@ export const useNavigateNextQuestById = (questId: Nullable<string>) => {
...
+  useEffect(() => {
+    if (netInfo.isConnected === false) {
+      showInternetErrorAlert(
+        t('errors.network_progress.title'),
+        t('errors.network_progress.description'),
+      );
+    }
+    // intentionally
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [netInfo.isConnected]);
```
Comment (Reykjavik151) [inline app/hooks/quest.tsx]: the same as above, destruct it correctly

File: app/hooks/quest.tsx [hunk 4]
```diff
@@ -101,6 +122,17 @@ export const useNavigateNextQuest = () => {
     state => state.quest.allQuests?.[currentLanguage ?? 'en'],
   );
 
+  useEffect(() => {
+    if (netInfo.isConnected === false) {
```
Comment (Reykjavik151) [inline app/hooks/quest.tsx]:
```suggestion
if (!isConnected)

File: app/hooks/quest.tsx [hunk 5]
```diff
@@ -101,6 +122,17 @@ export const useNavigateNextQuest = () => {
...
+      showInternetErrorAlert(
+        t('errors.network_progress.title'),
+        t('errors.network_progress.description'),
+      );
+    }
+    // intentionally
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [netInfo.isConnected]);
```
Comment (Reykjavik151) [inline app/hooks/quest.tsx]: the same as above

File: app/sagas/initQuest.ts [hunk 6]
```diff
@@ -5,26 +5,32 @@ import { IQuestDatoCms } from '../models/IQuest';
...
+
+      if (allCorrectQuests) {
+        yield put(questSlice.actions.saveAllQuestsSuccess(allCorrectQuests));
+      } else {
+        yield put(
+          questSlice.actions.saveAllQuestsError('get all quests error'),
+        );
+      }
     }
   }
```
Comment (Reykjavik151) [inline app/sagas/initQuest.ts:35]: what should we show if `isConnected` false?

File: app/screens/EditProfile/EditProfile/EditProfile.tsx [hunk 7]
```diff
@@ -18,22 +19,25 @@ import { useAppDispatch, useAppSelector } from '../../../hooks';
...
   const { type, userId } = route.params?.data;
 
+  const { t } = useTranslation();
   const dispatch = useAppDispatch();
+  const netInfo = useNetInfo();
 
   const parentData = useAppSelector(state => state.user.parent);
   const children = useAppSelector(state => state.user.children);
 
+  const isConnected = useMemo(() => netInfo.isConnected, [netInfo.isConnected]);
```
Comment (Reykjavik151) [inline app/screens/EditProfile/EditProfile/EditProfile.tsx]: the same as above

File: app/screens/EditProfile/EditProfile/EditProfile.tsx [hunk 8]
```diff
@@ -47,6 +51,13 @@ export const EditProfileScreen: React.FC<IEditProfileScreenProps> = ({
   const Icon = user && AVATARS_SVG[user?.avatar];
 
   const onSubmit = useCallback(() => {
+    if (isConnected === false) {
```
Comment (Reykjavik151) [inline app/screens/EditProfile/EditProfile/EditProfile.tsx:52]: show we do `=== false` here?

File: app/screens/Garden/Garden/Garden.tsx [hunk 9]
```diff
@@ -38,6 +39,18 @@ export const GardenScreen: React.FC<IGardenScreenProps> = ({
   const appStatus = useAppState();
   const dispatch = useAppDispatch();
   const isFocused = useIsFocused();
+  const netInfo = useNetInfo();
+
+  useEffect(() => {
+    if (netInfo.isConnected === false) {
```
Comment (Reykjavik151) [inline app/screens/Garden/Garden/Garden.tsx]: the same as above (naming, destructing, should we use `=== false`)

File: app/screens/Initial/InitialScreen.tsx [hunk 10]
```diff
@@ -23,12 +27,18 @@ export const InitialScreen: React.FC<IInitialScreenProps> = ({
...
     state => state.app.loading,
   );
 
+  const [isStartLoading, setIsStartLoading] = useState(false);
+
+  useEffect(() => {
+    if (netInfo.isConnected === false) {
+      showInternetErrorAlert();
+    }
+  }, [netInfo.isConnected]);
```
Comment (Reykjavik151) [inline app/screens/Initial/InitialScreen.tsx:41]: the same as above

File: app/screens/Menu/ChangePassword/ChangePassword.tsx [hunk 11]
```diff
@@ -20,25 +21,35 @@ import { ChangePasswordValidationSchema } from './ChangePassword.validation';
...
+      if (isConnected) {
+        dispatch(
+          userSlice.actions.changePassword({ newPassword, currentPassword }),
+        );
+        return;
+      }
+      showInternetErrorAlert(
+        t('errors.network.title'),
+        t('errors.network.description'),
       );
```
Comment (Reykjavik151) [inline app/screens/Menu/ChangePassword/ChangePassword.tsx:49]:
```suggestion
if (!isConnected) {
showError();
return;
}

dispatch

File: app/screens/Menu/DeleteAccount/DeleteAccount.tsx [hunk 12]
```diff
@@ -29,14 +31,24 @@ export const DeleteAccountScreen: React.FC<IDeleteAccountScreenProps> = ({
   const { t } = useTranslation();
 
   const dispatch = useAppDispatch();
+  const netInfo = useNetInfo();
+
+  const isConnected = useMemo(() => netInfo.isConnected, [netInfo.isConnected]);
```
Comment (Reykjavik151) [inline app/screens/Menu/DeleteAccount/DeleteAccount.tsx]: the same as above

File: app/screens/Menu/DeleteAccount/DeleteAccount.tsx [hunk 13]
```diff
@@ -29,14 +31,24 @@ export const DeleteAccountScreen: React.FC<IDeleteAccountScreenProps> = ({
...
     password => {
-      dispatch(userSlice.actions.deleteAccount(password));
+      if (isConnected) {
+        dispatch(userSlice.actions.deleteAccount(password));
+        return;
+      }
+      showInternetErrorAlert(
+        t('errors.network.title'),
+        t('errors.network.description'),
+      );
```
Comment (Reykjavik151) [inline app/screens/Menu/DeleteAccount/DeleteAccount.tsx:46]: the same as above

File: app/screens/NotificationAlert/NotificationAlert.tsx [hunk 14]
```diff
@@ -22,11 +24,11 @@ export const NotificationAlert: React.FC = ({}) => {
   useEffect(() => {
     if (currentError) {
       showMessage({
-        message: currentError,
+        message: t(currentError),
         style: {
           ...styles.container,
           marginTop: statusBarHeight.top,
-          paddingTop: -statusBarHeight.top,
+          paddingTop: -statusBarHeight.top + 16,
```
Comment (Reykjavik151) [inline app/screens/NotificationAlert/NotificationAlert.tsx:31]: Do not use magic numbers as `16`

File: app/screens/Reusable/ElixirTitleButton/ElixirTitleButton.tsx [hunk 15]
```diff
@@ -22,6 +24,23 @@ export const ElixirTitleButtonScreen: React.FC<IElixirTitleButtonScreenProps> =
...
+    useEffect(() => {
+      if (isConnected === false) {
+        showInternetErrorAlert(
+          t('errors.network_progress.title'),
+          t('errors.network_progress.description'),
+        );
+      }
+      // intentionally
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isConnected]);
```
Comment (Reykjavik151) [inline app/screens/Reusable/ElixirTitleButton/ElixirTitleButton.tsx]: create custom hook and reuse it acroos the app (for common scenario)

File: app/utils/index.ts [hunk 16]
```diff
@@ -168,3 +177,23 @@ export const LottieAbsoluteStyles = (
     ...styles,
   };
 };
+
+export const getIsConnected = async () =>
+  await NetInfo.fetch().then(state => state.isConnected);
```
Comment (Reykjavik151) [inline app/utils/index.ts]:
```suggestion
export const getIsConnected = async () => {
const netInfo = await NetInfo.fetch();
return netInfo.isConnected;
}

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

File: app/constants/error.ts (hunk 2 above)
Comment (emberglaz) [reply app/constants/error.ts:3]: when there is no internet, there are no translations

File: app/hooks/quest.tsx (hunk 4 above)
Comment (emberglaz) [reply app/hooks/quest.tsx]: can be null so is strictly false

File: app/sagas/initQuest.ts (hunk 6 above)
Comment (emberglaz) [reply app/sagas/initQuest.ts:35]: InitialScreen will show error

**Review by Reykjavik151 — APPROVED** [review]

**Pull request merged**